CdnEngine_Azure_MI.php

```diff
@@ -0,0 +1,414 @@
+<?php
+/**
+ * File: CdnEngine_Azure.php
+ *
+ * @package W3TC
+ */
+
+namespace W3TC;
+
+/**
+ * Windows Azure Storage CDN engine
+ */
+class CdnEngine_Azure_MI extends CdnEngine_Base {
+
+	/**
+	 * Constructor.
+	 *
+	 * @param array $config Configuration.
+	 */
+	public function __construct( $config = array() ) {
+		$config = array_merge(
+			array(
+				'user' => '',
+				'clientid' => getenv('ENTRA_CLIENT_ID')?: '',
+				'container' => '',
+				'cname' => array(),
+			),
+			$config
+		);
+
+		parent::__construct( $config );
+
+		// Load the Composer autoloader.
+		require_once W3TC_DIR . '/vendor/autoload.php';
+	}
+
+	/**
+	 * Inits storage client object
+	 *
+	 * @param string $error Error message.
+	 * @return bool
+	 */
+	public function _init( &$error ) {
+		if ( empty( $this->_config['user'] ) ) {
+			$error = 'Empty account name.';
+			return false;
+		}
+
+		if ( empty( $this->_config['clientid'] ) ) {
+			$this->_config['clientid'] = getenv('ENTRA_CLIENT_ID');
```

Comment:
this mean that it can be passed as Environment variable also? 

How as user will specify these Environment variables from Plug-in experience?

Reply:
This is an optional step that can help get the value from the env variable if the value is not available in the input config. Would be helpful with App Service if there is any issue while updating the plugin configuration.